c7n/resources/dynamodb.py

```diff
@@ -240,6 +240,138 @@ def process(self, resources):
                                 f.exception()))
 
 
+@Table.action_registry.register('create-backup')
+class CreateBackup(BaseAction, StatusFilter):
+    """Creates a manual backup of a DynamoDB table
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: dynamodb-create-backup
+                resource: dynamodb-table
+                actions:
+                  - type: create-backup
+    """
+
+    valid_status = ('ACTIVE',)
+    schema = type_schema('create-backup')
+    permissions = ('dynamodb:CreateBackup',)
+
+    def process(self, tables):
+        tables = self.filter_table_state(
+            tables, self.valid_status)
+        if not len(tables):
+            return
+
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for t in tables:
+                futures.append(w.submit(
+                    self.process_dynamodb_tables,
+                    tables))
+            for f in as_completed(futures):
+                if f.exception():
+                    self.log.error(
+                        "Exception creating dynamodb table backup \n %s",
+                        f.exception())
+        return tables
+
+    def process_dynamodb_tables(self, tables):
+
+        c = local_session(self.manager.session_factory).client('dynamodb')
+        for t in tables:
+            arn = c.create_backup(
+                BackupName=snapshot_identifier(
+                    'Backup', t['TableName']),
+                TableName=t['TableName'])['BackupDetails']['BackupArn']
+            t['c7n:BackupArn'] = arn
```

Comment:
we should also be copying tags over to the backup, I believe they are taggable but that would need to be verified. we also need to add a tag that custodian itself was performing the backup, so that when we gc we can also be specific to the custodian taken backups optionally.